### DISCUSSION TITLE
Convert existing gds to 3D format

### DISCUSSION THREAD

**Discussion by pruj-speed** [gdsfactory/gdsfactory#3330] [Q&A]
Hi,

Is gdsfactory a good tool to convert a gds layout into a 3D format like step? Any advice appreciated.

I notice slide 79 in the google doc with a cross section and a 3D rendering. It mentions tools --> pyxs

Is there a simple python example to help me get started?

Thanks!

**Comment by simbilod** [comment]
For STEP file output specifically, GMSH is able to output STEP (https://www.cfd-online.com/Forums/openfoam-meshing/226443-save-geometry-created-opencascade-step-file-gmsh.html). So using the meshing plugin this should be possible https://gdsfactory.github.io/gplugins/notebooks/meshing_05_3D.html, but you would need to adapt the code a bit

For other 3D formats, look into [component.to_3d](https://github.com/gdsfactory/gdsfactory/blob/2c30da587a3673a51c5babdd7f29ca92c6e38a35/gdsfactory/component.py#L934)
